static/components.ts

```diff
@@ -646,10 +648,12 @@ export function getOptPipelineViewWith(
         componentName: OPT_PIPELINE_VIEW_COMPONENT_NAME,
         componentState: {
             id,
+            lang,
+            compiler: compilerId,
             compilerName,
             editorid,
             treeid,
-            selectedFunction: '',
+            selectedGroup: '',
```

Comment:
Changing this will quite possibly break pre-existing links to opt viewers. We store the state and we need to make sure the state is backwards-compatible.

That also means we need to make sure the new `lang` and `compiler` fields can be guessed if not supplied, and should probably all be marked like "compiledId?: string` to note their optionality in the State object.

Reply:
Backwards compatibility support added, so I believe this is okay now.

Reply:
Thank you!